controllers/managedocs_controller.go

```diff
@@ -977,16 +1062,45 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 			Key: alertRelabelConfigSecretKey,
 		}
 
-		// prometheus tls -- will be uncommented
-		// tlsConfig := r.prometheus.Spec.Web.TLSConfig
-		// tlsConfig.KeySecret.LocalObjectReference.Name = prometheusServiceSecretName
-		// tlsConfig.Cert.Secret.LocalObjectReference.Name = prometheusServiceSecretName
-		// tlsConfig.ClientCA.ConfigMap.LocalObjectReference.Name = servingCertCABundleConfigMapName
+		if deployerCSV, err := r.getCSVByPrefix(deployerCSVPrefix); err == nil {
+			deployerCSVDeployment := &deployerCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0]
+			deployerCSVContainers := deployerCSVDeployment.Spec.Template.Spec.Containers
+			var kubeRbacImage string
+			for key := range deployerCSVContainers {
+				container := deployerCSVContainers[key]
+				if container.Name == "kube-rbac-proxy" {
+					kubeRbacImage = container.Image
+				}
+			}
+
+			prometheusContainers := r.prometheus.Spec.Containers
+			for key := range prometheusContainers {
+				container := &prometheusContainers[key]
+				if container.Name == "kube-rbac-proxy" {
+					container.Image = kubeRbacImage
+				}
+			}
+		} else {
+			return fmt.Errorf("Unable to set image for kube-rbac-proxy container: %v", err)
+		}
+
+		volumes := r.prometheus.Spec.Volumes
+		for key := range volumes {
+			volume := &volumes[key]
+			switch volume.Name {
+			case templates.ServingCertificateKeyVolumeName:
+				volume.VolumeSource.Secret.SecretName = prometheusServiceSecretName
+			case templates.ServiceCACertificateVolumeName:
+				volume.VolumeSource.ConfigMap.LocalObjectReference.Name = servingCertCABundleConfigMapName
+			case templates.KubeRbacProxyConfigVolumeName:
+				volume.VolumeSource.ConfigMap.LocalObjectReference.Name = kubeRbacConfigMap
+			}
+		}
```

Comment:
All of this is constant, no need to run code to set this up.
Please move this as static configuration into the Prometheus template